torch_geometric/nn/glob/gmt.py

```diff
@@ -0,0 +1,204 @@
+        out = out if getattr(self, 'ln1', None) is None else self.ln1(out)
+
+        return out
+
+
+class SAB(torch.nn.Module):
+    def __init__(self, dim_in, dim_out, num_heads,
+                 conv=None, ln=False):
+        super(SAB, self).__init__()
+        self.mab = MAB(dim_in, dim_in, dim_out, num_heads, conv=conv, ln=ln)
+
+    def forward(self, X, graph=None, attention_mask=None):
+        return self.mab(X, X, graph, attention_mask)
+
+
+class PMA(torch.nn.Module):
+    def __init__(self, dim, num_heads,
+                 num_seeds, conv=None, ln=False):
+        super(PMA, self).__init__()
+        self.S = torch.nn.Parameter(torch.Tensor(1, num_seeds, dim))
+        torch.nn.init.xavier_uniform_(self.S)
+
+        self.mab = MAB(dim, dim, dim, num_heads, conv=conv, ln=ln)
+
+    def forward(self, X, graph=None, attention_mask=None):
+        return self.mab(
+            self.S.repeat(X.size(0), 1, 1), X, graph, attention_mask
+        )
+
+
+class GraphMultisetTransformer(torch.nn.Module):
+    r"""The global pooling operator, Graph Multiset Transformer,
+    from the `"Accurate Learning of Graph Representations
+    with Graph Multiset Pooling" <https://arxiv.org/abs/2102.11533>`_ paper.
+
+    In short, nodes of the entire graph is clustered with attention-based
+    pooling operations (GMPool_G or GMPool_I), and self-attention (SelfAtt)
+    could be used to calculate the inter-relationships among nodes.
+
+    Args:
+        in_channels (int): Size of each input sample.
+        hidden_channels (int): Size of each hidden sample.
+        out_channels (int): Size of each output sample.
+        conv (torch.nn.Module, optional): A graph neural network layer
+            for calculating hidden representations of nodes for GMPool_G
+            (one of :class:`torch_geometric.nn.conv.GraphConv`,
+            :class:`torch_geometric.nn.conv.GCNConv` or
+            :class:`torch_geometric.nn.conv.GATConv`). (default:
+            :class:`torch_geometric.nn.conv.GCNConv`)
+        num_nodes (int, optional): The number of average
+            or maximum nodes. (default: :obj:`300`)
+        pooling_ratio (float, optional): Graph pooling ratio
+            for each pooling. (default: :obj:`0.25`)
+        pool_sequences ([str], optional): A sequence of pooling layers
+            consisting of Graph Multiset Transformer (one of
+            :obj:`['GMPool_I']`,
+            :obj:`['GMPool_G']`,
+            :obj:`['GMPool_G', 'GMPool_I']`,
+            :obj:`['GMPool_G', 'SelfAtt', 'GMPool_I']` or
+            :obj:`['GMPool_G', 'SelfAtt', 'SelfAtt', 'GMPool_I']`).
+            (default: :obj:`['GMPool_G', 'SelfAtt', 'GMPool_I']`)
+        num_heads (int, optional): Number of heads in attention.
+            (default: :obj:`4`)
+        ln (bool, optional): If set to :obj:`True`, will make use of
+            layer normalization. (default: :obj:`False`)
+    """
+    def __init__(self, in_channels, hidden_channels, out_channels,
```

Comment:
Can we add type hints? For example, `in_channels: int`, `hidden_channels: int`, ...

Reply:
Dear @rusty1s
I have added the type hints.

Thanks for the great suggestion!